front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy

```diff
@@ -82,13 +92,12 @@ class PipelineController {
   @PreAuthorize("@fiatPermissionEvaluator.storeWholePermission() and hasPermission(#pipeline.application, 'APPLICATION', 'WRITE') and @authorizationSupport.hasRunAsUserPermission(#pipeline)")
   @RequestMapping(value = '', method = RequestMethod.POST)
   void save(@RequestBody Pipeline pipeline) {
-    if (!pipeline.application || !pipeline.name) {
-      throw new InvalidEntityException("A pipeline requires name and application fields")
-    }
+
+    validatePipeline(pipeline)
+
     pipeline.name = pipeline.getName().trim()
 
     if (!pipeline.id) {
-      checkForDuplicatePipeline(pipeline.getApplication(), pipeline.getName())
```

Comment:
Moved this into the validate call, this will still check for duplicate names in the case of the pipeline id being set which seems like it would be an update case.